### PR TITLE
SUB-2424 reusable python ut workflow

### PR DESCRIPTION
**Pull request by jchima**

### CODE REVIEW

File: .github/workflows/nodejs-ut-ci.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: npm ci
+      #- run: npm run build --if-present
+      #- run: npm run prettier
+      - run: npm run lint
+      - run: npm test
+  # code-analysis:
```
Comment (tevans-submittable) [inline .github/workflows/nodejs-ut-ci.yml:32]: I dont think you meant to include this in this branch, it was probably created from the npm branch and not master.  Do you want to abandon the other PR and work both files here?

File: .github/workflows/nodejs-ut-ci.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+        #Choose which node version [10.x, 12.x, 14.x, 15.x]. Default is 15.x
+          node-version: 15.x
+      
+      - name: Run Tests
+        run: npm ci
+      - run: npm run build --if-present
```
Comment (tevans-submittable) [inline .github/workflows/nodejs-ut-ci.yml:28]: Lets make sure each step has a name

File: .github/workflows/nodejs-ut-ci.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    #Enable only when Sonar server is available. It will run Static Code Analysis
+  #     - name: Run Sonar Scan
+  #       uses: sonarsource/sonarqube-scan-action@master
+  #     env:
+  #       SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+  #       SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+
+  #     - name: Bump version and push tag
+  #       id: tag_version
+  #       uses: anothrNick/github-tag-action@1.35.0
```
Comment (tevans-submittable) [inline .github/workflows/nodejs-ut-ci.yml:40]: Were are using our own fork of this now right?

File: .github/workflows/python-ut.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
Comment (tevans-submittable) [inline .github/workflows/python-ut.yml:32]: Do we really need to call flake8 twice?

File: .github/workflows/nodejs-ut-ci.yml (hunk 2 above)
Comment (jchima) [reply .github/workflows/nodejs-ut-ci.yml]: Got it, name added

File: .github/workflows/python-ut.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      #     GITHUB_TOKEN: ${{ secrets.GITHUBUSER }} #create a github token and secrets
+      #   with:
+      #     tag_name: ${{ steps.tag_version.outputs.new_tag }}
+      #     release_name: ${{ steps.tag_version.outputs.new_tag }}
+      #     body: ${{ steps.tag_version.outputs.changelog }}
+      
+      # Sends a slack notification to slack build channel on failure
+      - name: Report Status
+        if: endsWith(GitHub.ref, 'main') && failure()
+        uses: ravsamhq/notify-slack-action@v1
```
Comment (tevans-submittable) [inline .github/workflows/python-ut.yml:73]: Switch to our repo

File: .github/workflows/python-ut.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        with:
+          status: ${{ job.status }}
+          notify_when: 'failure'
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
+
+      # Sends a slack notification to slack build channel
+      - name: Slack Notification
+        if: endsWith(GitHub.ref, 'main') && success()
+        uses: rtCamp/action-slack-notify@v2.2.0
```
Comment (tevans-submittable) [inline .github/workflows/python-ut.yml:83]: Switch to our repo

File: .github/workflows/nodejs-ut-ci.yml (hunk 3 above)
Comment (jchima) [reply .github/workflows/nodejs-ut-ci.yml]: oh yes we're, updated

File: .github/workflows/python-ut.yml (hunk 5 above)
Comment (jchima) [reply .github/workflows/python-ut.yml]: Done

File: .github/workflows/python-ut.yml (hunk 6 above)
Comment (jchima) [reply .github/workflows/python-ut.yml]: Done

File: .github/workflows/python-ut.yml (hunk 4 above)
Comment (jchima) [reply .github/workflows/python-ut.yml]: Updated, please check

**Review by tevans-submittable — APPROVED** [review]
LGTM

**Pull request merged**